mpsl/include/mpsl_temp.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2011 - 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @defgroup mpsl_temp_api MPSL Library Temp API
+ * @{
+ *
+ * @brief APIs for the MPSL Temp module.
+ *
+ */
+
+#ifndef MPSL_TEMP_H__
+#define MPSL_TEMP_H__
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+#include <stdint.h>
+
+
+/** @brief Get the temperature measured on the chip
+ *
+ * This function will block until the temperature measurement is done.
+ * It takes around 50 us from call to return.
+ *
+ * @retval Result of temperature measurement. Die temperature in 0.25 degrees Celsius.
+ */
+int32_t mpsl_temperature_get(void);
```

Comment:
How do CONFIG_MPSL and CONFIG_TEMP_NRF5 integrate?

Should we enforce through Kconfig that they can not be enabled simultaneously?

e.g. in MPSL have 'depends on ! TEMP_NRF5'

Reply:
Hmm yes, I suppose so. I haven't looked much into this, but do you mean to add a driver for mpsl to implement the sensor.h interface @joerchan ? I.e. sensor_sample_fetch()?
I can add a 'depends on ! TEMP_NRF5' for MPSL for now, and then for the next MPSL release we can consider implementing an mpsl temp driver?

Reply:
I get a dependency loop when I add the 'depends on !TEMP_NRF5'.. I think it's because CLOCK_CONTROL_NRFXLIB selects CLOCK_CONTROL_NRF_FORCE_ALT, and if !CLOCK_CONTROL_NRF_FORCE_ALT, CLOCK_CONTROL_NRF_USES_TEMP_SENSOR defaults to y if (...) and selects TEMP_NRF5. So there is a loop there..

Reply:
Perhaps we can have

```
if(CONFIG_TEMP_NRF5 AND CONFIG_MPSL)
message(WARNING "CONFIG_TEMP_NRF5 and CONFIG_MPSL have both been enabled, but are not designed to co-exist")
endif()

```
in the mpsl build scripts.

Reply:
Good idea.

Reply:
see https://github.com/NordicPlayground/nrfxlib/pull/93/files#diff-c9acbd21266be1e5ada0464309e8ceaeR18